File: crates/suspect-codegen/src/java_sdk/NativeCredentialEnvOpenRouter.java
import ai.openrouter.sdk.*;
import ai.openrouter.sdk.Client.*;
import java.util.*;

/** Actual OpenRouter schemas; JDK transport capture, never real account requests. */
public final class NativeCredentialEnvOpenRouter {
    private NativeCredentialEnvOpenRouter() {}
    private static final String KEY="{\"data\":{\"label\":\"controlled\",\"limit\":null,\"limit_remaining\":null,\"limit_reset\":null,\"usage\":1e-400,\"usage_daily\":0,\"usage_weekly\":0,\"usage_monthly\":0,\"byok_usage\":0,\"byok_usage_daily\":0,\"byok_usage_weekly\":0,\"byok_usage_monthly\":0,\"is_free_tier\":false,\"is_management_key\":false,\"is_provisioning_key\":false,\"include_byok_in_limit\":false,\"creator_user_id\":null,\"rate_limit\":{\"requests\":-1,\"interval\":\"none\",\"note\":\"controlled\"}}}";
    private static void check(boolean value,String message){if(!value)throw new AssertionError(message);}
    private static String authorization(NativeSupport.Mock transport){return transport.requests.getLast().headers().entrySet().stream().filter(e->e.getKey().equalsIgnoreCase("Authorization")).map(e->e.getValue().getFirst()).findFirst().orElse("");}
    public static void main(String[] args)throws Exception{
        var key=new NativeSupport.Mock(new NativeSupport.Script(200,"application/json",KEY));
        try(var client=Client.fromEnv(key)){
            var result=client.getCurrentKey();check(result.status()==200&&!result.data().data().isManagementKey()&&result.data().data().limit()==null&&result.data().data().usage().token().equals("1e-400"),"actual current-key schema decoding");
            check(key.requests.getLast().method().equals("GET")&&key.requests.getLast().uri().equals("https://openrouter.ai/api/v1/key"),"source-default HTTPS GET /key");check(authorization(key).equals("Bearer controlled-openrouter-token"),"source apiKey is bearer");
        }
        var credits=new NativeSupport.Mock(new NativeSupport.Script(200,"application/json","{\"data\":{\"total_credits\":100.25,\"total_usage\":1.5}}"));
        try(var client=Client.fromEnv(credits)){
            var result=client.getCreditsAsync().get();check(result.status()==200&&result.data().data().totalCredits().token().equals("100.25"),"actual optional credits schema");check(credits.requests.getLast().uri().equals("https://openrouter.ai/api/v1/credits"),"source-default optional management path");
        }
        var missing=new NativeSupport.Mock(new NativeSupport.Script(200,"application/json",KEY));
        try(var client=new Client(HttpRuntime.Options.builder().httpClient(missing).build())){try{client.getCurrentKey();throw new AssertionError("explicit empty options used environment");}catch(SdkException error){check(error.kind().equals("missing-credential")&&missing.requests.isEmpty()&&error.capture().length==0&&error.getCause()==null,"missing key did not fail before HTTPS");}}
        var denied=new NativeSupport.Mock(new NativeSupport.Script(401,"application/json","{\"error\":{\"code\":401,\"message\":\"denied\"}}"));
        try(var client=Client.fromEnv(denied)){try{client.getCurrentKey();throw new AssertionError("401 became success");}catch(GetCurrentKeyStatus401 error){check(error.status()==401&&error.data().error().message().equals("denied")&&!error.toString().contains("controlled-openrouter-token"),"typed current-key failure leaked credential");}}
        check(key.shutdowns.get()==0&&credits.shutdowns.get()==0&&denied.shutdowns.get()==0,"caller transport ownership");
        System.out.println("JAVA_CREDENTIAL_ENV_OPENROUTER_OK: compiled ai.openrouter.sdk.Client.fromEnv; source-default GET /key and optional /credits; typed decode and secret-free pre-HTTP missing credentials");
    }
}

File: crates/suspect-codegen/src/dart_sdk/native_credential_env_openrouter.dart
import 'dart:async';
import 'dart:convert' show utf8;
import 'package:openrouter/openrouter.dart';

const keyReply='{"data":{"label":"controlled fixture","limit":null,"usage":0.125,"usage_daily":0,"usage_weekly":0,"usage_monthly":0,"byok_usage":0,"byok_usage_daily":0,"byok_usage_weekly":0,"byok_usage_monthly":0,"is_free_tier":false,"is_management_key":false,"is_provisioning_key":false,"limit_remaining":null,"limit_reset":null,"include_byok_in_limit":false,"creator_user_id":null,"rate_limit":{"requests":-1,"interval":"unused","note":"controlled fixture"}}}';
const creditsReply='{"data":{"total_credits":9007199254740993.0000000001,"total_usage":0.125}}';
void check(bool condition,String message){if(!condition)throw StateError(message);}
final class Capture implements HttpTransport {
  Capture(this.token);
  final String token;int requests=0;int releases=0;
  @override Future<TransportResponse> send(TransportRequest request) async {
    check(request.method=='GET','source method');
    check(request.url.toString()=='https://openrouter.ai/api/v1/key'||request.url.toString()=='https://openrouter.ai/api/v1/credits','source-default HTTPS URL');
    check(request.headers['authorization']=='Bearer $token','source bearer attachment from runtime value');requests++;
    return TransportResponse(status:200,headers:const {'content-type':['application/json']},body:Stream.value(utf8.encode(request.url.path.endsWith('/key')?keyReply:creditsReply)),onClose:() async {releases++;});
  }
  @override Future<void> close() async {check(requests==releases,'release');}
}
Future<void> calls(Client client) async {
  final key=await client.getCurrentKey();check(key.status==200&&key.data.data.usage.token=='0.125'&&!key.data.data.isManagementKey,'decoded actual key schema');
  final credits=await client.getCredits();check(credits.status==200&&credits.data.data.totalCredits.token=='9007199254740993.0000000001','decoded actual credits schema');
}
Future<void> realEnvironment() async {
  final transport=Capture('native-openrouter');final client=Client(transport:transport);
  try {await calls(client);}finally{await client.close();}
  print('DART_OPENROUTER_ENV_VM_OK');
}
Future<void> unavailable() async {
  final transport=Capture('never');final client=Client(transport:transport);
  try {try {await client.getCurrentKey();throw StateError('missing env accepted');}on ConfigurationException catch(error){check(error.message.length<256&&!error.toString().contains('node-env-must-not-be-read'),'bounded secret-free missing error');}check(transport.requests==0,'before HTTP');}
  finally{await client.close();}
  final explicitTransport=Capture('explicit-openrouter');final explicit=Client(transport:explicitTransport,credentials:const Credentials(apiKey:'explicit-openrouter'),environment:(_)=>throw StateError('must not read'));
  try {await calls(explicit);}finally{await explicit.close();}
  print('DART_OPENROUTER_ENV_UNAVAILABLE_OK');
}
Future<void> main() async {
  await unavailable();
  var reads=0;var value='portable-openrouter';final transport=Capture(value);
  final client=Client(transport:transport,environment:(name){check(name=='OPENROUTER_API_KEY','configured variable name');reads++;return value;});
  value='changed';try {await calls(client);check(reads==1,'portable creation-time snapshot');}finally{await client.close();}
  print('DART_OPENROUTER_ENV_PORTABLE_OK');
}
